docs/src/content/docs/components/Utility/Navigation.mdx

```diff
@@ -0,0 +1,390 @@
+nav.updateAction('select', {
+    key: {binds: ['SPACEBAR'], type: ['press']},
+    button: {binds: ['face-button-down'], type: 'press'},
+    callback: (scope) => handleSelection(scope),
+    global: true
+});
+```
+
+### Adding Custom Actions
+
+Beyond the default actions, you can add your own custom actions for game-specific inputs:
+
+```tsx
+import Navigation from '@components/Navigation/Navigation/Navigation';
+import { ActionMap } from '@components/Navigation/Navigation/types';
+
+const customActions: ActionMap = {
+    // Custom action for opening inventory
+    'open-inventory': {
+        key: {binds: ['I'], type: ['press']},
+        button: {binds: ['face-button-top'], type: 'press'},
+        callback: () => openInventory(),
+        global: true
+    },
+    // Custom action for opening map
+    'open-map': {
+        key: {binds: ['M'], type: ['press']},
+        button: {binds: ['face-button-left'], type: 'press'},
+        callback: (scope) => openMap(scope),
+        global: false  // Only emit locally, not globally
+    }
+};
+
+<Navigation actions={customActions}>
+    {/* Your UI */}
+</Navigation>
+```
+
+Or add them programmatically:
+
+```tsx
+const nav = useNavigation();
+
+nav.addAction('toggle-menu', {
+    key: {binds: ['TAB'], type: ['press']},
+    button: {binds: ['START'], type: 'press'},
+    callback: () => toggleMenu(),
+    global: true
+});
+```
+
+### Subscribing to Actions in Custom Components
+
+Custom components can respond to navigation actions using the `navigationActions` directive from `BaseComponent`.
+This directive automatically listens for action events and executes callbacks when the component (or its anchor) is focused.
+
+To subscribe your component to an action, follow these steps:
+
+<Steps>
+
+1. Import the `useBaseComponent` hook in your component
+
+2. Extract the `navigationActions` method from the hook
+
+3. Add the `navigationActions` method as an attribute to the element you wish to subscribe, and prefix it with the `use:` keyword
+
+4. Provide the names of the actions you wish to subscribe to and their corresponding callbacks (what will happen when they're triggered)
+
+</Steps>
+
+```tsx
+import { useBaseComponent } from '@components/BaseComponent/BaseComponent';
+const MyComponent = (props) => {
+    const { navigationActions } = useBaseComponent(props);
+    return (
+        <div use:navigationActions={{
+            select: () => console.log('Item selected'),
+            back: () => console.log('Going back')
+        }}>
+        </div>
+    );
+};
+```
+
+:::caution
+Only actions with `global: true` (including all default actions) can be subscribed to using the `navigationActions` directive. Actions with `global: false` only execute their callbacks and don't emit events for components to listen to.
+:::
+
+**With an anchor element:**
+
+Sometimes you want a component to respond to actions when a different element is focused (like a child button):
```

Comment:
Can we add more descriptive example - for example how to propagate action when a row with a dropdown is focused. The dropdown should receive the actions from the row via the anchor.